### PR TITLE
[test] Add concurrency and power scenarios to chip-level tests

### PR DESCRIPTION
**Pull request by msfschaffner**
This adds concurrency and power scenario tests identified during the AST testplanning.

This also covers the power-related tests drafted here: #9297

Let me know if additional scenarios should be added, or whether the test descriptions should be updated.

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 1]
```diff
@@ -2983,5 +2983,121 @@
...
+          - aon_timer active
+          - adc_ctrl active in low power mode
+          - TBD: check whether sysrst_ctrl and pinmux wakeup detectors should be active
+          - TBD: check whether PWM should be active
+
+      This test should leverage the OTTF test framework for supporting
+      concurrency in a FreeRTOS environment.
+      '''
+      stage: V2
+      tests: []
```
Comment (tjaychen) [inline hw/top_earlgrey/data/chip_testplan.hjson:3100]: for this particular one, do you want to x-ref the tests the tests that are already done?

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 2]
```diff
@@ -2983,5 +2983,121 @@
...
+      entry and exit scenarios.
+
+      The test should be made configurable so that the type of power state and
+      the time spent in a particular power state can be configured via a
+      flag (or similar). This will make it easier to reuse the test for power
+      simulation and characterization later on.
+
+      Blocks / functionality to run simulatenously in this test:
+
+        - Staggered activation of OTBN, aes, KMAC/HMAC.
```
Comment (tjaychen) [inline hw/top_earlgrey/data/chip_testplan.hjson:3020]: is the text below referenced from @timothytrippel power virus issue already?

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 3]
```diff
@@ -2983,5 +2983,121 @@
...
+      desc: '''Concurrency test modeling load conditions in idle state
+
+      This concurrency test models an average idle scenarios.
+
+      The test should be made configurable so that the type of power state and
+      the time spent in a particular power state can be configured via a
+      flag (or similar). This will make it easier to reuse the test for power
+      simulation and characterization later on.
+
+      The test should cover the following scenarios:
```
Comment (tjaychen) [inline hw/top_earlgrey/data/chip_testplan.hjson:3059]: for both this test and the test below, i would recommend the test itself toggle a GPIO / pin to indicate when the state of interest has begun.  This will make it for Nuvoton to follow along on when exactly should measure.

A log statement in the test would be helpful too.

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 4]
```diff
@@ -2983,5 +2983,121 @@
       stage: V2
       tests: []
     }
+    {
+      name: chip_sw_ast_concurrency
+      desc: '''Concurrency test with several AST functionalities enabled.
```
Comment (tjaychen) [inline hw/top_earlgrey/data/chip_testplan.hjson:2988]: can this particular one be merged into `max_load`?

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 2 above)
Comment (msfschaffner) [reply hw/top_earlgrey/data/chip_testplan.hjson:3020]: Are you referring to [this doc](https://docs.google.com/document/d/1OMnxqSZ0OGNzHkWH8Le6utW6llyIR1QgrUnbZpOkwYU/edit#
) here?

I took the description for this test point mainly from https://github.com/lowRISC/opentitan/issues/9297 since it spells out the scenarios quite nicely (the doc talks more about implementation steps needed for the test). I can however make sure to add a reference to the doc here (e.g. via #14095) or/and by adding a link to the corresponding chip-level test issue I am about to create.

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 4 above)
Comment (msfschaffner) [reply hw/top_earlgrey/data/chip_testplan.hjson:2988]: yeah we should be able to absorb that. let me rephrase this.

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 3 above)
Comment (msfschaffner) [reply hw/top_earlgrey/data/chip_testplan.hjson:3059]: Ack. We may also want to add this to the max load test above.

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 1 above)
Comment (msfschaffner) [reply hw/top_earlgrey/data/chip_testplan.hjson:3100]: Yeah this should be x'refd with tests that are already completed.

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 2 above)
Comment (tjaychen) [reply hw/top_earlgrey/data/chip_testplan.hjson:3020]: o that would be great! thank you!
i know i wrote the original steps, but i can't remember if people actually agreed to them..

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 3 above)
Comment (tjaychen) [reply hw/top_earlgrey/data/chip_testplan.hjson:3059]: yes good point.

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 1 above)
Comment (tjaychen) [reply hw/top_earlgrey/data/chip_testplan.hjson:3100]: sounds good.

**Review by tjaychen — APPROVED** [review]

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 2 above)
Comment (msfschaffner) [reply hw/top_earlgrey/data/chip_testplan.hjson:3020]: Yeah I think we all agree with them. 
The doc actually references your initial issue regarding the combination of blocks that should be active.

Comment (msfschaffner) [reply hw/top_earlgrey/data/chip_testplan.hjson:3005]: Ok I added a reference to https://github.com/lowRISC/opentitan/issues/14095 in the description.

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 3 above)
Comment (msfschaffner) [reply hw/top_earlgrey/data/chip_testplan.hjson:3049]: Done!

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 4 above)
Comment (msfschaffner) [reply hw/top_earlgrey/data/chip_testplan.hjson]: Rephrased and combine (one testpoint less, yay!)

File: hw/top_earlgrey/data/chip_testplan.hjson (hunk 1 above)
Comment (msfschaffner) [reply hw/top_earlgrey/data/chip_testplan.hjson:3092]: Which test would be a good one to extend here? `chip_sw_pwrmgr_deep_sleep_all_wake_ups`?

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 5]
```diff
@@ -2983,5 +2983,115 @@
...
+          - TBD: check whether PWM should be active
+
+      This test should leverage the OTTF test framework for supporting
+      concurrency in a FreeRTOS environment. See also the design docs linked
+      in #14095 for more details on how to approach the implementation.
+
+      X'ref'ed with power manager sleep / wakeup test.
+      '''
+      stage: V2
+      tests: ["chip_sw_pwrmgr_deep_sleep_all_wake_ups"]
```
Comment (msfschaffner) [inline hw/top_earlgrey/data/chip_testplan.hjson:3094]: FYI

**Comment by sriyerg** [comment]
minor nit- most testpoints have been limited to 100 chars / line, though it's not strictly enforced.

**Comment by msfschaffner** [comment]
@moidx @timothytrippel @arunthomas some of these fall into the category of "virus" tests for the chip-level.
